sunpy/map/sources/rhessi.py

```diff
@@ -46,12 +44,13 @@ def __init__(self, data, header, **kwargs):
         # Fix some broken/misapplied keywords
         if self.meta['ctype1'] == 'arcsec':
             self.meta['cunit1'] = 'arcsec'
-            self.meta['ctype1'] = 'HPLN-TAN'
+            self._fix_and_warn('cunit1', 'arcsec', replace_old=True)
+            self._fix_and_warn('ctype1', 'HPLN-TAN', replace_old=True)
         if self.meta['ctype2'] == 'arcsec':
-            self.meta['cunit2'] = 'arcsec'
-            self.meta['ctype2'] = 'HPLT-TAN'
+            self._fix_and_warn('cunit2', 'arcsec', replace_old=True)
```

Comment:
haven't we already injected this above?

Reply:
I've moved this editing to before `super().__init__` is called, and only warned for replacing `ctype` and not `cunit`.